lib/phonebook/class/complete_member.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+import 'package:myecl/phonebook/class/membership.dart';
+import 'member.dart';
+
+class CompleteMember {
+  CompleteMember({
+    required this.member,
+    required this.memberships,
+  });
+
+  late final Member member;
+  late final List<Membership> memberships;
+
+  CompleteMember.fromJson(Map<String, dynamic> json) {
+    debugPrint(json.toString());
+    member = Member(
+        name: json['name'],
+        firstname: json['firstname'],
+        nickname: json['nickname'],
+        id: json['id'],
+        email: json['email'],
+        promotion: json['promotion']??"");
+    memberships = List<Membership>.from(json['memberships']
+        .map((membership) => Membership.fromJson(membership)));
+        
+  }
+
+  Map<String, dynamic> toJson() {
+    final data = <String, dynamic>{
+      'member': member.id,
+      'memberships': memberships.map((e) => e.toJson()).toList(),
+    };
+    return data;
+  }
+
+  CompleteMember copyWith({
+    Member? member,
+    List<Membership>? membership,
+  }) {
+    return CompleteMember(
+      member: member ?? this.member,
+      memberships: membership ?? memberships,
+    );
+  }
+
+  CompleteMember.empty() {
+    member = Member.empty();
+    memberships = [];
+  }
+
+  Member toMember() {
+    return Member(
+      name: member.name,
+      firstname: member.firstname,
+      nickname: member.nickname,
+      id: member.id,
+      email: member.email,
+      promotion: member.promotion,
+    );
+  }
+
+  @override
+  String toString() {
+    return 'CompleteMember(member: $member, memberships: $memberships)';
```

Comment:
Print every parameter please :)

Reply:
A complete member is composed of a member and a list of membership, so all attributes are printed with this line